guess-combo-game.js

```diff
@@ -1,229 +1 @@
-function makeGuess(lowRange, highRange) {
-  guess = Math.floor(Math.random() * (highRange - lowRange) + lowRange);
-  return guess;
-}
-
-// Make guess at number in range based on binary search technique 
-function makeOptimizedGuess(lowRange, highRange) {
-  lowRange = parseInt(lowRange);
-  highRange = parseInt(highRange);
-  return guess = Math.floor((lowRange + highRange) / 2);
-}
-
-// Provide hint of whether secret number is higher or lower than current guess
-function giveHint(guess, secret) {
-  if (guess < secret) {
-    return 'higher';
-  }
-  else if (guess > secret) {
-    return 'lower';
-  }
-  else {
-    return null;
-  }
-}
-
-// Check to see if human is giving correct hint based on secret number
-function humanIsCheating(hint, guess, secret) {
-  if (hint.toLowerCase() === 'l'
-    && (guess <= secret)) {
-    return true;
-  }
-  else if (hint.toLowerCase() === 'h'
-    && (guess >= secret)) {
-    return true;
-  }
-  else {
-    return false;
-  }
-}
-
-// Start game and allow player to choose variation they wish to play
-async function startGame() {
-
-  console.log("\n\nLet's play a number guessing game.");
-  console.log("There are two variations of the game: \n");
-  console.log("1 - You pick a number and the computer tries to guess it.");
-  console.log("2 - You and the computer take turns until one of you guesses it.\n");
-
-  while (true) {
-    gameSelection = await ask("Enter a 1 or 2 to pick the game to play: ");
-
-    switch (gameSelection) {
-      case '1':
-        console.log('\n\n');
-        console.log('*************************************************');
-        console.log('** Computer guesses a number that you pick!    **');
-        console.log('*************************************************\n');
-        computerGuesses();
-        break;
-
-      case '2':
-        console.log('\n\n');
-        console.log('*************************************************');
-        console.log('**  Human & Computer try to guess the number!! **');
-        console.log('*************************************************\n');
-        humanAndComputerGuess();
-        break;
-
-      default:
-        console.log("You must enter a 1 or a 2 to select a game.");
-        continue;
-    }
-  }
-}
-
-// Computer and Human Play
-async function humanAndComputerGuess() {
-  let lowPoint = 1;
-  let highPoint = null;
-  let computerLowPoint = lowPoint;
-  let computerHighPoint = null;
-
-  // Ask for high point of guess range
-  while (true) {
-    highPoint = await ask("What number would you like for the high point of the guess range? ");
-    if (highPoint > 1) {
-      computerHighPoint = highPoint;
-      break;
-    }
-    else {
-      console.log("The high point of the guess range must be greater than 1");
-    }
-  }
-
-  // Generate secret number
-  let secretNumber = makeGuess(lowPoint, highPoint);
-  //console.log('The secret number is: ' + secretNumber + '\n');
-
-  while (true) {
-
-    while (true) {
```

Comment:
You could condense and refactor some of your while loops as they are a little redundant in some of your code, like these two for example. However, by going through your other files, I can see a sense of refactor along other aspects of your code so I'm sure as the class  progress you'll continue this trend.